src/plugins/Multipart.js

```diff
@@ -177,9 +170,12 @@ module.exports = class Multipart extends Plugin {
     //   }
   }
 
-  handleUpload () {
+  handleUpload (fileIDs) {
     this.core.log('Multipart is uploading...')
-    const files = this.core.getState().files
+    const files = fileIDs.map(getFile, this)
+    function getFile (fileID) {
+      return this.core.state.files[fileID]
+    }
```

Comment:
All changes in this file are because the file IDs to upload are now passed in by `Core`, so the Multipart plugin doesn't have to figure out what it has to upload.